src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -3160,8 +3195,21 @@ var projectUniqueNamesForBuildIntToUpdate
 
                 // Write out the lock file, now no need bubbling re-evaluating of parent projects when you restore from PM UI.
                 // We already taken account of that concern in PreviewBuildIntegratedProjectsActionsAsync method.
+
+                bool isNoOp = projectAction.RestoreResultPair.Result is NoOpRestoreResult;
+                IReadOnlyList<string> filesToBeUpdated = isNoOp ? null : GetFilesToBeUpdated(projectAction.RestoreResultPair);
+                if (!isNoOp)
+                {
+                    RestoreProgressReporter?.StartProjectUpdate(projectAction.RestoreResultPair.SummaryRequest.Request.Project.FilePath, filesToBeUpdated);
+                }
+
                 await RestoreRunner.CommitAsync(projectAction.RestoreResultPair, token);
 
+                if (!isNoOp)
```

Comment:
Can the user cancel the operation and cause `CommitAsync()` to throw?  If so, should this be in a `try...finally`?

Reply:
Technically yes, so let's handle it. 

Fixed in the latest commit.